plugins/in_proc/in_proc.c

```diff
@@ -521,6 +484,41 @@ static int in_proc_exit(void *data, struct flb_config *config)
     return 0;
 }
 
+static struct flb_config_map config_map[] = {
+    {
+      FLB_CONFIG_MAP_INT, "interval_sec", DEFAULT_INTERVAL_SEC,
+      0, FLB_TRUE, offsetof(struct flb_in_proc_config, interval_sec),
+      "Set the collector interval"
+    },
+    {
+      FLB_CONFIG_MAP_INT, "interval_nsec", DEFAULT_INTERVAL_NSEC,
+      0, FLB_TRUE, offsetof(struct flb_in_proc_config, interval_nsec),
+      "Set the collector interval (sub seconds)"
+    },
+    {
+     FLB_CONFIG_MAP_BOOL, "alert", "false",
+     0, FLB_TRUE, offsetof(struct flb_in_proc_config, alert),
+     "Only generate alerts if process is down"
+    },
+    {
+     FLB_CONFIG_MAP_BOOL, "mem", "true",
+     0, FLB_TRUE, offsetof(struct flb_in_proc_config, mem),
+     "Append memory usage to record"
+    },
+    {
+     FLB_CONFIG_MAP_BOOL, "fd", "true",
+     0, FLB_TRUE, offsetof(struct flb_in_proc_config, fds),
+     "Append fd count to record"
+    },
+    {
+     FLB_CONFIG_MAP_STR, "proc_name", "",
```

Comment:
The default value should be NULL.
Because `in_proc` will illegal output if there is no `proc_name` and default is "".

The output of pid is negative.
```
[0] proc.0: [1645567433.915966114, {"alive"=>false, "proc_name"=>"", "pid"=>-1, "mem.VmPeak"=>0, "mem.VmSize"=>128, "mem.VmLck"=>0, "mem.VmHWM"=>94429484768116, "mem.VmRSS"=>140701048282640, "mem.VmData"=>140701142938593, "mem.VmStk"=>124554051584, "mem.VmExe"=>8, "mem.VmLib"=>140701136894336, "mem.VmPTE"=>8, "mem.VmSwap"=>140701048282640, "fd"=>0}]
```

```
$ bin/fluent-bit -i proc -o stdout
Fluent Bit v1.9.0
* Copyright (C) 2015-2021 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/02/23 07:03:52] [ info] [engine] started (pid=23512)
[2022/02/23 07:03:52] [ info] [storage] version=1.1.6, initializing...
[2022/02/23 07:03:52] [ info] [storage] in-memory
[2022/02/23 07:03:52] [ info] [storage] normal synchronization mode, checksum disabled, max_chunks_up=128
[2022/02/23 07:03:52] [ info] [cmetrics] version=0.2.3
[2022/02/23 07:03:52] [ info] [sp] stream processor started
[2022/02/23 07:03:52] [ info] [output:stdout:stdout.0] worker #0 started
[0] proc.0: [1645567433.915966114, {"alive"=>false, "proc_name"=>"", "pid"=>-1, "mem.VmPeak"=>0, "mem.VmSize"=>128, "mem.VmLck"=>0, "mem.VmHWM"=>94429484768116, "mem.VmRSS"=>140701048282640, "mem.VmData"=>140701142938593, "mem.VmStk"=>124554051584, "mem.VmExe"=>8, "mem.VmLib"=>140701136894336, "mem.VmPTE"=>8, "mem.VmSwap"=>140701048282640, "fd"=>0}]
[1] proc.0: [1645567434.916434153, {"alive"=>false, "proc_name"=>"", "pid"=>-1, "mem.VmPeak"=>0, "mem.VmSize"=>140701136894336, "mem.VmLck"=>16, "mem.VmHWM"=>15, "mem.VmRSS"=>140701048273720, "mem.VmData"=>140701142938593, "mem.VmStk"=>140701136894384, "mem.VmExe"=>8, "mem.VmLib"=>140701136894336, "mem.VmPTE"=>8, "mem.VmSwap"=>140701048272948, "fd"=>0}]
[2] proc.0: [1645567435.916707421, {"alive"=>false, "proc_name"=>"", "pid"=>-1, "mem.VmPeak"=>0, "mem.VmSize"=>140701136894336, "mem.VmLck"=>43265, "mem.VmHWM"=>15, "mem.VmRSS"=>140701048273720, "mem.VmData"=>140701142938593, "mem.VmStk"=>140701136894384, "mem.VmExe"=>8, "mem.VmLib"=>140701136894336, "mem.VmPTE"=>8, "mem.VmSwap"=>140701048272948, "fd"=>0}]
[3] proc.0: [1645567436.917727857, {"alive"=>false, "proc_name"=>"", "pid"=>-1, "mem.VmPeak"=>0, "mem.VmSize"=>140701136894336, "mem.VmLck"=>32514, "mem.VmHWM"=>15, "mem.VmRSS"=>140701048273720, "mem.VmData"=>140701142938593, "mem.VmStk"=>140701136894384, "mem.VmExe"=>8, "mem.VmLib"=>140701136894336, "mem.VmPTE"=>8, "mem.VmSwap"=>140701048272948, "fd"=>0}]
^C[2022/02/23 07:03:58] [engine] caught signal (SIGINT)
[0] proc.0: [1645567437.916194081, {"alive"=>false, "proc_
```

Reply:
Could you add testcase to check blank `proc_name` ?